e2e_playwright/st_markdown.py

```diff
@@ -116,3 +82,98 @@
 
 """
 )
+
+
+# Headers in markdown tests (originally from the typography-test suite).
+
+with st.container():
+    st.markdown("# some really long header " + " ".join(["lol"] * 10))
+    st.markdown(
+        """
+| Col1      | Col2        | Col3        | Col4        |
+| --------- | ----------- | ----------- | ----------- |
+| Some      | Data        | Data        | Data        |
+"""
+    )
+
+
+def draw_header_test(join_output):
+    strings = [
+        "# Header header1",
+        "## Header header2",
+        "### Header header3",
+        "#### Header header4",
+        "##### Header header5",
+        "###### Header header6",
+        "Quisque vel blandit mi. Fusce dignissim leo purus, in imperdiet lectus suscipit nec.",
+    ]
+
+    if join_output:
+        st.markdown("\n\n".join(strings))
+    else:
+        for string in strings:
+            st.markdown(string)
+
+
+draw_header_test(True)
+
+with st.sidebar:
+    st.text_input("This is a label", key="1")
+    draw_header_test(True)
+
+"---"
+
+with st.container():
+    st.text("Headers in single st.markdown command")
+    draw_header_test(True)
+
+"---"
+
+with st.container():
+    st.text("Headers in multiple st.markdown command")
+    draw_header_test(False)
+
+"---"
+
+with st.container():
+    st.text("Headers in columns")
+
+    a, b = st.columns(2)
+
+    with a:
+        draw_header_test(True)
+
+    with b:
+        draw_header_test(False)
+
+"---"
+
+with st.container():
+    st.text("Headers in columns with other elements above")
+
+    a, b = st.columns(2)
+
+    with a:
+        st.text("This is some text")
+        draw_header_test(True)
+
+    with b:
+        st.text("This is some text")
+        with st.container():
+            draw_header_test(False)
+
+"---"
+
+with st.container():
+    st.text("Headers in column beside widget")
+
+    a, b = st.columns(2)
+
+    with a:
+        st.write("# Header header")
+        st.write("## Header header")
+
+    with b:
+        st.text_input("This is a label", key="2")
+
+"---"
```

Comment:
This is the existing typography test code that is migrated in this PR from Cypress